src/cdomains/floatDomain.ml

```diff
@@ -83,33 +85,12 @@ end
 
 module FloatIntervalImpl(Float_t : CFloatType) = struct
   include Printable.Std (* for default invariant, tag and relift *)
-  type t = (Float_t.t * Float_t.t) option [@@deriving eq, ord, to_yojson, hash]
-
-  let to_int ik = function
-    | None -> IntDomain.IntDomTuple.top_of ik
-    (* special treatment for booleans as those aren't "just" truncated *)
-    | Some (l, h) when ik = IBool && (l > Float_t.zero || h < Float_t.zero) -> IntDomain.IntDomTuple.of_bool IBool true
-    | Some (l, h) when ik = IBool && l = h && l = Float_t.zero -> IntDomain.IntDomTuple.of_bool IBool false
-    | Some (l, h) when ik = IBool -> IntDomain.IntDomTuple.top_of IBool
-    | Some (l, h) -> 
-      (* as converting from float to integer is (exactly) defined as leaving out the fractional part,
-         (value is truncated towrad zero) we do not require specific rounding here *)
-      IntDomain.IntDomTuple.of_interval ik (Float_t.to_big_int l, Float_t.to_big_int h)
-
-  let of_int x = 
-    match IntDomain.IntDomTuple.minimal x, IntDomain.IntDomTuple.maximal x with
-    | Some l, Some h when l >= Float_t.to_big_int Float_t.lower_bound && h <= Float_t.to_big_int Float_t.upper_bound ->
-      let l' = Float_t.of_float Down (Big_int_Z.float_of_big_int l) in
-      let h' = Float_t.of_float Up (Big_int_Z.float_of_big_int h) in
-      if not (Float_t.is_finite l' && Float_t.is_finite h') then
-        None
-      else
-        Some (l', h')
-    | _, _ -> None
+  type t = Top | Bot | Interval of (Float_t.t * Float_t.t) [@@deriving eq, ord, to_yojson, hash]
 
   let show = function
-    | None -> Float_t.name ^ ": [Top]"
-    | Some (low, high) -> Printf.sprintf "%s:[%s,%s]" Float_t.name (Float_t.to_string low) (Float_t.to_string high)
+    | Top -> Float_t.name ^ ":[Top]"
```

Comment:
When merging, you will have to remove the `Float_t.name` here as it is now added on the layer above in order to also handle long doubles